advancedblacklist/commands/blacklist.py

```diff
@@ -0,0 +1,171 @@
+            elif user.id == ctx.guild.owner_id:
+                return await ctx.send("You cannot add the guild owner to the local blacklist.")
+            elif ctx.author == user:
+                return await ctx.send("You cannot add yourself to the local blacklist.")
+        reason = reason or "No reason provided."
+        await add_to_blacklist(self.bot, users, reason, guild=ctx.guild)
+        that = "that user" if len(users) == 1 else "those users"
+        await ctx.send(f"Done. Added {that} to the local blacklist.")
+
+    @local_blacklist.command(
+        name="remove", aliases=["del", "delete", "rm"], require_var_positional=True
+    )
+    async def local_blacklist_remove(self, ctx: commands.Context, *users: discord.Member):
+        """Remove users from the local blacklist.
+
+        **Arguments**
+            - `users` The users to remove from the local blacklist.
+        """
+        if not await get_blacklist(self.bot, ctx.guild):
+            return await ctx.send("There are no users in the local blacklist.")
+        await remove_from_blacklist(self.bot, users, guild=ctx.guild)
+        that = "that user" if len(users) == 1 else "those users"
+        await ctx.send(f"Done. Removed {that} from the local blacklist.")
+
+    @local_blacklist.command(name="reason")
+    async def local_blacklist_reason(
+        self, ctx: commands.Context, user: discord.Member, *, reason: str
+    ):
+        """Edit the reason for a user in the local blacklist.
+
+        **Arguments**
+            - `user` The user to edit the reason of. This cannot be a bot.
+            - `reason` The reason for blacklisting the user.
+        """
+        if await self.bot.is_owner(user):
+            return await ctx.send("That user is a bot owner.")
+        elif ctx.guild.owner_id == user.id:
+            return await ctx.send("That user is the guild owner.")
+        elif user.bot:
+            return await ctx.send("That user is a bot.")
+        elif not await in_blacklist(self.bot, user.id, ctx.guild):
+            return await ctx.send("That user is not locally blacklisted.")
+        try:
+            await edit_reason(self.bot, user, reason, False, guild=ctx.guild)
+        except KeyError:
+            return await ctx.send("That user was not in the local blacklist.")
+        await ctx.send("The reason for that user has been edited.")
+
+    @local_blacklist.command(name="list")
+    async def local_blacklist_list(self, ctx: commands.Context):
+        """List the users in the local blacklist."""
+        if not (bl := await get_blacklist(self.bot, ctx.guild)):
+            return await ctx.send("There are no users in the local blacklist.")
+        msg = "Locally Blacklisted Users:"
+        for key, value in bl.items():
+            name = u.name if (u := self.bot.get_user(key)) else "Unknown User"
+            msg += f"\n\t- [{key}] {name}: {value}"
+        await ctx.send_interactive(pagify(msg, page_length=1800), "yml")
```

Comment:
```suggestion
        if not members:
            raise commands.UserInputError
        for member in members:
            if member.bot:
                return await ctx.send("You cannot add bots to the local blacklist.")
            elif await self.bot.is_owner(member ):
                return await ctx.send("You cannot add a bot owner to the local blacklist.")
            elif member.id == ctx.guild.owner_id:
                return await ctx.send("You cannot add the guild owner to the local blacklist.")
            elif ctx.author == member:
                return await ctx.send("You cannot add yourself to the local blacklist.")
        reason = reason or "No reason provided."
        await add_to_blacklist(self.bot, members, reason, guild=ctx.guild)
        that = "that member" if len(users) == 1 else "those members"
        await ctx.send(f"Done. Added {that} to the local blacklist.")

    @local_blacklist.command(
        name="remove", aliases=["del", "delete", "rm"], require_var_positional=True
    )
    async def local_blacklist_remove(self, ctx: commands.Context, *members: discord.Member):
        """Remove members from the local blacklist.

        **Arguments**
            - `members` The members to remove from the local blacklist.
        """
        if not await get_blacklist(self.bot, ctx.guild):
            return await ctx.send("There are no members in the local blacklist.")
        await remove_from_blacklist(self.bot, members, guild=ctx.guild)
        that = "that member" if len(users) == 1 else "those members"
        await ctx.send(f"Done. Removed {that} from the local blacklist.")

    @local_blacklist.command(name="reason")
    async def local_blacklist_reason(
        self, ctx: commands.Context, member: discord.Member, *, reason: str
    ):
        """Edit the reason for a member in the local blacklist.

        **Arguments**
            - `member` The user to edit the reason of. This cannot be a bot.
            - `reason` The reason for blacklisting the member.
        """
        if await self.bot.is_owner(member):
            return await ctx.send("That member is a bot owner.")
        elif ctx.guild.owner_id == member.id:
            return await ctx.send("That member is the guild owner.")
        elif member.bot:
            return await ctx.send("That member is a bot.")
        elif not await in_blacklist(self.bot, member.id, ctx.guild):
            return await ctx.send("That memberis not locally blacklisted.")
        try:
            await edit_reason(self.bot, member, reason, False, guild=ctx.guild)
        except KeyError:
            return await ctx.send("That member was not in the local blacklist.")
        await ctx.send("The reason for that member has been edited.")

    @local_blacklist.command(name="list")
    async def local_blacklist_list(self, ctx: commands.Context):
        """List the members in the local blacklist."""
        if not (bl := await get_blacklist(self.bot, ctx.guild)):
            return await ctx.send("There are no members in the local blacklist.")
        msg = "Locally Blacklisted members:"
        for key, value in bl.items():
            name = u.name if (u := self.bot.get_user(key)) else "Unknown Member"
            msg += f"\n\t- [{key}] {name}: {value}"
        await ctx.send_interactive(pagify(msg, page_length=1800), "yml")
```
Worth seeing my review comment above about this. Also updated variable names to keep consistency within the code.